v2/util/ip.go

```diff
@@ -73,6 +73,19 @@ func IsIANAReserved(ip net.IP) bool {
 	return false
 }
 
+// IntersectsIANAReserved checks if a CIDR intersects any IANA reserved CIDRs
+func IntersectsIANAReserved(net net.IPNet) bool {
+	if !net.IP.IsGlobalUnicast() {
+		return true
+	}
+	for _, reserved := range reservedNetworks {
+		if reserved.Contains(net.IP) || net.Contains(reserved.IP) {
```

Comment:
Does this handle the case where `net.IP == reserved.IP`?

Reply:
Any two networks that have the same base number should contain each other, [example](https://play.golang.org/p/tO-H4Mr6cWs).